sqlite_test.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestOpenDB(t *testing.T) {
+	db := openTestDB(t)
+	var journalMode string
+	if err := db.QueryRow("PRAGMA journal_mode;").Scan(&journalMode); err != nil {
+		t.Fatal(err)
+	}
+	if want := "wal"; journalMode != want {
+		t.Errorf("journal_mode=%q, want %q", journalMode, want)
+	}
+	var synchronous string
+	if err := db.QueryRow("PRAGMA synchronous;").Scan(&synchronous); err != nil {
+		t.Fatal(err)
+	}
+	if want := "0"; synchronous != want {
+		t.Errorf("synchronous=%q, want %q", synchronous, want)
+	}
+	if err := db.Close(); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func openTestDB(t *testing.T) *sql.DB {
+	t.Helper()
+	db, err := sql.Open("sqlite3", "file:"+t.TempDir()+"/test.db")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := db.Exec("PRAGMA journal_mode=WAL"); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := db.Exec("PRAGMA synchronous=OFF"); err != nil {
+		t.Fatal(err)
+	}
+	return db
+}
+
+// execContexter is an *sql.DB or an *sql.Tx.
+type execContexter interface {
```

Comment:
Could use driver.ExecerContext instead, which afaict is the same interface.

Reply:
That appears to return a `driver.Result`, not an `sql.Result`.